services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java

```diff
@@ -344,19 +346,20 @@ private Response buildAuthorizationCodeAuthorizationResponse() {
     }
 
     private Response buildRegister() {
-        authManager.expireIdentityCookie(session);
+        AuthenticationManager.expireIdentityCookie(session);
 
         AuthenticationFlowModel flow = realm.getRegistrationFlow();
         String flowId = flow.getId();
 
         AuthenticationProcessor processor = createProcessor(authenticationSession, flowId, LoginActionsService.REGISTRATION_PATH);
         authenticationSession.setClientNote(APP_INITIATED_FLOW, LoginActionsService.REGISTRATION_PATH);
+        LocaleUtil.processLocaleParam(session, realm, authenticationSession);
```

Comment:
This is the actual fix as it propagates the value from the KC_LOCALE_PARAM param to the proper auth session note.

Reply:
The other two changes were just minor cleanups to get rid of warnings.

Reply:
@thomasdarimont - can you please amend an existing test or write a new test for this, so we don't break it in a future change? Thanks!

Reply:
I minimized the change and added a small test.